src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,73 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
 public class WorkWithFile {
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String COMMA = ",";
+    private static final String RESULT = "result";
+    private static final int OPERATION_INDEX = 0;
+    private static final int AMOUNT_INDEX = 1;
+    private static final int DEFAULT_AMOUNT = 0;
+    private static final int REQUIRED_PARTS_COUNT = 2;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        try {
+            Map<String, Integer> operationTotals = readData(fromFileName);
+            int supplyTotal = operationTotals.getOrDefault(SUPPLY, DEFAULT_AMOUNT);
+            int buyTotal = operationTotals.getOrDefault(BUY, DEFAULT_AMOUNT);
+            int result = supplyTotal - buyTotal;
+            writeReport(toFileName, supplyTotal, buyTotal, result);
+        } catch (Exception e) {
+            System.err.println("Error processing files: " + e.getMessage());
+            e.printStackTrace();
```

Comment:
Leaving empty catch block or e.printStackTrace here is a bad practice. Better re-throw RuntimeException with original exception and informative message in the parameters:

```
catch (Exception e) {
    throw new RuntimeException("Can't read data from the file " + fileName, e);
}
```